### ISSUE TITLE
failure on thread/test_async.ctx_timer_block/0

### ISSUE THREAD

**Issue by alinask** [openucx/ucx#2765]
```
13:49:30 [ RUN      ] thread/test_async.ctx_timer_block/0
13:49:30 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node/worker/1/contrib/../test/gtest/ucs/test_async.cc:516: Failure
13:49:30 Expected: (lt.count()) >= (1), actual: 0 vs 1
13:49:30 
[  FAILED  ] thread/test_async.ctx_timer_block/0, where GetParam() = 1 (45 ms)
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7372/label=hpc-test-node,worker=1/consoleFull

**Comment by evgeny-leksikov** [comment]
Seems like related failure:
```
19:43:35 [ RUN      ] thread/test_async.ctx_timer/0
19:43:35 /scrap/jenkins/workspace/hpc-ucx-pr/label/r-vmb-ppc-jenkins/worker/0/contrib/../test/gtest/ucs/test_async.cc:449: Failure
19:43:35 Expected: (lt.count()) >= (COUNT / 4), actual: 9 vs 10
19:43:35 
[  FAILED  ] thread/test_async.ctx_timer/0, where GetParam() = 1 (199 ms)
```

**Comment by dmitrygladkov** [comment]
```
15:23:37 [----------] 4 tests from thread_spinlock/test_async
15:23:37 [ RUN      ] thread_spinlock/test_async.ctx_event/0
15:23:37 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node/worker/1/contrib/../test/gtest/ucs/test_async.cc:451: Failure
15:23:37 Expected: (le.count()) >= (1), actual: 0 vs 1
15:23:38 
[  FAILED  ] thread_spinlock/test_async.ctx_event/0, where GetParam() = 1 (247 ms)
```
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10746/label=hpc-test-node,worker=1/consoleFull

@yosefe should we reopen this isssue?

**Comment by yosefe** [comment]
Fixed by #3754
